esx_service/vmdk_ops.py

```diff
@@ -786,9 +786,23 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     vm_datastore_url = vmdk_utils.get_datastore_url_from_config_path(config_path)
     vm_datastore = get_datastore_name(vm_datastore_url)
 
+    # Detach is implicity allowed if a disk has been attached
+    # Do not find tenant or privileges; just detach the disk
+    if cmd == "detach":
```

Comment:
Good suggestion, but outside of scope for this PR. @shaominchen  - Please file an issue (or code it up and submit a PR) 

Generally, executeRequest() became too messy. It was a small function with a switch for 5 simple commands, now it's quite complex. Would be nice to clean up. Certainly not in this PR